src/tests/no-incompatible-branch-types.test.coffee

```diff
@@ -0,0 +1,68 @@
+path = require 'path'
+
+{ESLintUtils} = require '@typescript-eslint/experimental-utils'
+
+{rules: {'no-incompatible-branch-types': rule}} = require '..'
+
+ruleTester = new ESLintUtils.RuleTester(
+  parser: '@typescript-eslint/parser'
+  parserOptions:
+    project: './tsconfig.eslint.json'
+    tsconfigRootDir: path.join __dirname, '../../src/tests/fixtures'
```

Comment:
Sort of messy but the (new) `fixtures/` directory doesn't get copied from `src/` -> `lib/` so this has to point back from the transpiled `lib/` into `src/`